File: pset2/caesar/caesar.c
// Encrypt a word using Caesar's cipher.
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <cs50.h>

int main(int argc, string argv[])
{
    // Ensure the arg integer is inputted, otherwise return an error and exit.
    if (argc != 2)
    {
        printf("Error, please input an non-negative integer.\n");
        return 1;
    }

    int key = atoi(argv[1]);
    // Ensure the arg integer is non-negative, else return an error and exit.
    if (key < 0)
    {
        printf("Error, please input an non-negative integer.\n");
        return 1;
    }
    // Wrap the key if it goes past 26.
    key = key % 26;
    string plaintext = get_string("plaintext: ");
    printf("ciphertext: ");
    for (int i = 0; i < strlen(plaintext); i++)
    {
        if (isalpha(plaintext[i]))
        {
            char plainchar = plaintext[i];
            if (isupper(plainchar))
                plainchar = (((plainchar + key) - 65 ) % 26) + 65;
            if (islower(plainchar))
                plainchar = (((plainchar + key) - 97 ) % 26) + 97;
            plaintext[i] = plainchar;
        }
        printf("%c", plaintext[i]);
    }
    printf("\n");
    return 0;
}

File: pset6/caesar/caesar.py
# Port of caesar.c to Python
import cs50
from sys import argv

# Ensure the user entered at least one arg, otherwise exit
if (len(argv) != 2):
    print("Usage: python caesar.py <integer>")
    exit(1)

# init a key and wrap it around if it go past 26th letter
key = int(argv[1])
key = key % 26

# ask for a plaintext to cypher
print("plaintext: ", end="")
plaintext = cs50.get_string()

# output encrypted text using key
print("ciphertext: ", end="")
for c in plaintext:
    if c.isalpha():
        if c.isupper():
            c = chr((((ord(c) + key) - 65 ) % 26 ) + 65)
        if c.islower():
            c = chr((((ord(c) + key) - 97 ) % 26 ) + 97)
    print(c, end="")
print()